File: algorithms/2_medium/N/01343_number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold/solution.c
int numOfSubarrays(const int *restrict arr, const int len, const int k, const int threshold) {
    if (k <= 0 || threshold < 0) { return -1; }

    if (k > len) { return 0; }

    const int target = threshold * k;
    int cnt = 0;
    int sum = 0;
    for (int i = 0; i < k; i += 1) { sum += arr[i]; }

    if (sum >= target) { cnt += 1; }

    for (int i = k; i < len; i += 1) {
        sum = sum + arr[i] - arr[i - k];

        if (sum >= target) { cnt += 1; }
    }

    return cnt;
}

File: algorithms/2_medium/N/01343_number_of_sub_arrays_of_size_k_and_average_greater_than_or_equal_to_threshold/solution.rs
impl Solution {
    pub fn num_of_subarrays(arr: Vec<i32>, k: i32, threshold: i32) -> i32 {
        if k <= 0 || threshold < 0 { return -1; }

        let k = k as usize;
        let len = arr.len();
        if k > len { return 0; }

        let target = threshold * (k as i32);
        let mut cnt = 0_i32;
        let mut sum: i32 = arr[0..k].iter().sum();

        if sum >= target { cnt += 1; }

        for i in k..len {
            sum = sum + arr[i] - arr[i - k];

            if sum >= target { cnt += 1; }
        }

        cnt
    }
}
